### PR TITLE
CI Contract Build examples with external ADIOS

### PR DESCRIPTION
**Pull request by spyridon97**
CI Contract: Build examples with external ADIOS

Fixes https://github.com/ornladios/ADIOS2/issues/3795.
Fixes: #3811

### CODE REVIEW

**Comment by vicentebolea** [comment]
Re-add me as reviewer when ready to review :)

**Comment by vicentebolea** [comment]
Looks good, there is still some build/test error. Once #3915 is merged please rebase on the top of master so that we can add shellcheck coverage to this new shell scripts.

**Comment by spyridon97** [comment]
@vicentebolea I see something very weird for the examples contract

At the install step i see:

```
-- Installing: /__w/ADIOS2/ADIOS2/install/bin/adios2_hello_helloWorld
-- Set runtime path of "/__w/ADIOS2/ADIOS2/install/bin/adios2_hello_helloWorld" to ""
```

And at the test step i see:

```
+ /__w/ADIOS2/ADIOS2/install/bin/adios2_hello_helloworld
/opt/adios2/source/testing/contract/examples/test.sh: line 8: /__w/ADIOS2/ADIOS2/install/bin/adios2_hello_helloworld: No such file or directory
```

Am I missing something or that's wrong?

**Comment by vicentebolea** [comment]
> -- Set runtime path of "/__w/ADIOS2/ADIOS2/install/bin/adios2_hello_helloWorld" to ""


Wrong case in the W in helloworld

**Comment by spyridon97** [comment]
> > -- Set runtime path of "/__w/ADIOS2/ADIOS2/install/bin/adios2_hello_helloWorld" to ""
> 
> Wrong case in the W in helloworld

I fixed it

File: .shellcheck_exclude_paths [hunk 1]
```diff
@@ -15,6 +15,12 @@ scripts/developer/setup.sh
 scripts/docker/setup-user.sh
 scripts/runconf/runconf.sh
 scripts/runconf/runconf_olcf.sh
+testing/contract/examples/build.sh
+testing/contract/examples/config.sh
+testing/contract/examples/depends.sh
+testing/contract/examples/install.sh
+testing/contract/examples/setup.sh
+testing/contract/examples/test.sh
```
Comment (vicentebolea) [inline .shellcheck_exclude_paths:23]: We want to remove this so this newly shell scripts are covered by shellcheck

```suggestion
```

**Comment by vicentebolea** [comment]
@spyridon97 rebase to get rid of the CI unrelated errors.

**Comment by spyridon97** [comment]
> @spyridon97 rebase to get rid of the CI unrelated errors.

Can you approve now?

**Comment by vicentebolea** [comment]
> Can you approve now?

No, You still have CI failures

**Comment by spyridon97** [comment]
> > Can you approve now?
> 
> No, You still have CI failures

It should be passing now. I Fixed many but i had to surpress some warnings, cause i could not find a way to fix them.

File: testing/contract/examples/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -x
+set -e
+
+# shellcheck disable=SC1091
+source "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"/setup.sh
+
+# shellcheck disable=SC2154
```
Comment (vicentebolea) [inline testing/contract/examples/build.sh:9]: We need to address this warning, there are workarounds proposed for this in https://www.shellcheck.net/wiki/SC2154.

Comment (spyridon97) [reply testing/contract/examples/build.sh]: The listed suggestions did not help, but i found another solution, let me know what you think.

File: testing/contract/examples/install.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -x
+set -e
+
```
Comment (vicentebolea) [inline testing/contract/examples/install.sh:5]:
```suggestion
# Fail if build_dir is not set
${build_dir:?}
```

How about something like this?

Comment (spyridon97) [reply testing/contract/examples/install.sh:5]: Shell check says it's not set, so that will fail, right?

Comment (vicentebolea) [reply testing/contract/examples/install.sh:5]: It is one of the suggested fixes for the warning, https://www.shellcheck.net/wiki/SC2154. It fixes the warning in my machine

**Comment by spyridon97** [comment]
@vicentebolea shall we merge now?

**Review by vicentebolea — APPROVED** [review]
Looks great! Thanks @spyridon97

**Pull request merged**